code/dynamic_programming/src/fibonacci/fibonacci.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import "fmt"
+
+// Fibonacci function to calculate nth fibonacci number 
+func Fibonacci(n int) int {
+	a, b := 0, 1
+	fibonacci := 0
+	if n == 0 {
+		fibonacci = 0
+	} else if n == 1 {
+		fibonacci = 1
+	} else {
+		for i := 2; i <= n; i++ {
+			// Storing result in 2 varibles instead of an array
+			fibonacci = a + b
+			a = b
+			b = fibonacci
+		}
+	}
+	return fibonacci
+}
+
+func main() {
+	fmt.Println(Fibonacci(10)) // Output: 55
```

Comment:
instead of printing the values, can you please use `assert` statements?
Also please add a separate unit test `test_fibonacci.go`
```
import "github.com/stretchr/testify/assert"

assert(test, 55, Fibonacci(10))
```